identity/backend/src/main/java/io/camunda/identity/rolemanagement/service/RoleMembershipService.java

```diff
@@ -0,0 +1,127 @@
+import io.camunda.identity.usermanagement.service.UserService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class RoleMembershipService {
+
+  private final UserService userService;
+  private final GroupService groupService;
+  private final CamundaUserDetailsManager camundaUserDetailsManager;
+  private final RoleRepository roleRepository;
+
+  public RoleMembershipService(
+      final UserService userService,
+      final GroupService groupService,
+      final CamundaUserDetailsManager camundaUserDetailsManager,
+      final RoleRepository roleRepository) {
+    this.userService = userService;
+    this.groupService = groupService;
+    this.camundaUserDetailsManager = camundaUserDetailsManager;
+    this.roleRepository = roleRepository;
+  }
+
+  public List<String> getRolesOfUserByUserId(final long userId) {
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    return camundaUserDetails.getRoles();
+  }
+
+  public void assignRoleToUser(final String roleName, final long userId) {
+    if (!roleRepository.existsById(roleName)) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    final List<String> roles = new ArrayList<>();
+    roles.addAll(camundaUserDetails.getRoles());
+    roles.add(roleName);
+
+    saveUserDetailsWithRoles(camundaUserDetails, roles);
+  }
+
+  public void unassignRoleFromUser(final String roleName, final long userId) {
+    if (!roleRepository.existsById(roleName)) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    final CamundaUserDetails camundaUserDetails =
+        camundaUserDetailsManager.loadUserByUsername(
+            userService.findUserById(userId).getUsername());
+
+    final List<String> roles = new ArrayList<>();
+    roles.addAll(
+        camundaUserDetails.getRoles().stream().filter(rn -> !rn.equals(roleName)).toList());
+
+    saveUserDetailsWithRoles(camundaUserDetails, roles);
+  }
+
+  public List<String> getRolesOfGroupByGroupId(final Long groupId) {
+    final CamundaGroup camundaGroup = groupService.findGroupById(groupId);
+
+    return camundaUserDetailsManager.findGroupAuthorities(camundaGroup.name()).stream()
+        .map(GrantedAuthority::getAuthority)
+        .map(authorityName -> authorityName.replace("ROLE_", ""))
```

Comment:
❓ I'm a little worried at how brittle this could be, we will need to remember to do this everywhere, have you thought about controlling this at a root level with Spring config https://stackoverflow.com/questions/38134121/how-do-i-remove-the-role-prefix-from-spring-security-with-javaconfig/44828767#44828767?

Reply:
I thought about removing the prefix and dropped the thought maybe a bit too early on a second attempt now ... 

The issue is that we have the concept of roles and permissions and them being separate things. Spring however does blend them all together into the user's `authorities` giving roles a special prefix. By that, it can then internally decide which elements of a user's `authorities` are used to evaluate `hasRole(...)` ... 

In general a `hasRole("FUBAR")` and a `hasAuthority("ROLE_FUBAR")` are the same thing. 

In the end we would need to:
- make sure role names and permission names never collide ... which is achievable, but could be fiddly, depending on where we put that logic -> that worries me a bit, but might be less of a worry than the current prefixed thing ... 
- it will need us to give up some built-in logic around building users ... although I'm fairly certain this can be made work w/o too much issues ... 

@maryarm ... you've evaluated the whole built-in functionality, I'd love to hear your thoughts on this! 😃

Reply:
After looking into potential other use cases of "permissions" aka authorities, it should be fine if they are just prefixed. We can filter them internally and I don't see potential problems where (custom user provided) permission could have a name like `ROLE_xxx` .

Reply:
Alright then, I'll leave it this way for now. Thanks!